corehq/form_processor/tests/test_basics.py

```diff
@@ -500,6 +503,60 @@ def test_caching_form_attachment_during_submission(self):
         with patch.object(get_blob_db(), 'get', side_effect=Exception('unexpected blobdb read')):
             _submit_case_block(True, uuid.uuid4().hex, user_id='user2', update={})
 
+    def test_failed_form_with_case_with_secret(self):
+        with override_settings(XFORM_PRE_PROCESSORS={
+            DOMAIN: ['corehq.apps.data_vault.tests.test_utils.TestVaultPatternExtractor']}
+        ):
+            case_id = uuid.uuid4().hex
+            modified_on = datetime.utcnow()
+            self.assertEqual(VaultEntry.objects.count(), 0)
+
+            _submit_case_block(
+                True, case_id, user_id='user1', owner_id='owner1', case_type='demo',
+                case_name='this is a very long case name that exceeds the 255 char limit' * 5,
+                date_modified=modified_on, date_opened=modified_on, update={
+                    'dynamic': '123', 'secret_case_property': '7777777777'
+                }
+            )
+            with self.assertRaises(CaseNotFound):
+                self.casedb.get_case(case_id)
+            self.assertEqual(VaultEntry.objects.count(), 1)
+            vault_entry = VaultEntry.objects.last()
+            self.assertEqual(vault_entry.value, "7777777777")
+
+    def test_successful_form_with_case_with_secret(self):
+        with override_settings(XFORM_PRE_PROCESSORS={
+            DOMAIN: ['corehq.apps.data_vault.tests.test_utils.TestVaultPatternExtractor']}
+        ):
+            self.assertEqual(VaultEntry.objects.count(), 0)
+            case_id = uuid.uuid4().hex
+            modified_on = datetime.utcnow()
+            # create case
+            _submit_case_block(
+                True, case_id, user_id='user1', owner_id='owner1', case_type='demo',
+                case_name='secret_case', date_modified=modified_on, date_opened=modified_on, update={
+                    'dynamic': '123', 'secret_case_property': '0123456789'
+                }
+            )
+            case = self.casedb.get_case(case_id)
+            self.assertEqual(VaultEntry.objects.count(), 1)
+            vault_entry = VaultEntry.objects.last()
+            self.assertEqual(vault_entry.value, "0123456789")
+            self.assertEqual(case.get_case_property('secret_case_property'), f"vault:{vault_entry.key}")
+
+            # update case
+            _submit_case_block(
+                False, case_id, user_id='user2', owner_id='owner2',
+                case_name='update_secret_case', date_modified=modified_on, date_opened=modified_on, update={
+                    'secret_case_property': '9876543210'
+                }
+            )
+            case = self.casedb.get_case(case_id)
+            self.assertEqual(VaultEntry.objects.count(), 2)
+            vault_entry = VaultEntry.objects.last()
+            self.assertEqual(vault_entry.value, "9876543210")
+            self.assertEqual(case.get_case_property('secret_case_property'), f"vault:{vault_entry.key}")
```

Comment:
should probably also validate the form XML

Reply:
I did test for a successfull case in the other test, so skipped it here but sounds like worth adding. will do.

Reply:
I see, why are the tests separated? I think you should try put them all in their own test file.

Reply:
Tests here ensure correct case updates (like case property) whereas the other ones are around form processing and check for the form only (like form xml saved).